src/SAML2/XML/saml/Condition.php

```diff
@@ -8,61 +8,62 @@
 use SimpleSAML\Assert\Assert;
 use SimpleSAML\SAML2\Constants as C;
 use SimpleSAML\SAML2\Utils;
+use SimpleSAML\SAML2\XML\ExtensionPointInterface;
+use SimpleSAML\SAML2\XML\ExtensionPointTrait;
 use SimpleSAML\XML\Exception\InvalidDOMElementException;
+use SimpleSAML\XML\Exception\SchemaViolationException;
 
-use function trim;
+use function count;
+use function explode;
 
 /**
  * SAML Condition data type.
  *
  * @package simplesamlphp/saml2
  */
-abstract class Condition extends AbstractConditionType
+abstract class Condition extends AbstractConditionType implements ExtensionPointInterface
 {
-    /** @var string */
-    public const LOCALNAME = 'Condition';
+    use ExtensionPointTrait;
 
     /** @var string */
-    protected string $type;
-
-
-    /**
-     * Initialize a saml:Condition from scratch
-     *
-     * @param string $content
-     * @param string $type
-     */
-    protected function __construct(
-        string $content,
-        string $type
-    ) {
-        $this->setContent($content);
-        $this->setType($type);
-    }
+    public const LOCALNAME = 'Condition';
 
 
     /**
-     * Get the type of this Condition (expressed in the xsi:type attribute).
+     * Convert an XML element into a Condition.
      *
-     * @return string
-     */
-    public function getType(): string
-    {
-        return $this->type;
-    }
-
-
-    /**
-     * Set the type of this Condition (in the xsi:type attribute)
+     * @param \DOMElement $xml The root XML element
+     * @return \SimpleSAML\SAML2\XML\saml\Condition The condition
      *
-     * @param string $type
+     * @throws \SimpleSAML\XML\Exception\InvalidDOMElementException if the qualified name of the supplied element is wrong
      */
-    protected function setType(string $type): void
+    public static function fromXML(DOMElement $xml): object
     {
-        Assert::notWhitespaceOnly($type, 'The "xsi:type" attribute of a Condition cannot be empty.');
-        Assert::contains($type, ':');
-
-        $this->type = $type;
+        Assert::same($xml->localName, 'Condition', InvalidDOMElementException::class);
+        Assert::same($xml->namespaceURI, C::NS_SAML, InvalidDOMElementException::class);
+        Assert::true(
+            $xml->hasAttributeNS(C::NS_XSI, 'type'),
+            'Missing required xsi:type in <saml:Condition> element.',
+            SchemaViolationException::class
+        );
+
+        $type = $xml->getAttributeNS(C::NS_XSI, 'type');
+        Assert::validQName($type, SchemaViolationException::class);
+
+        $qname = explode(':', $type, 2);
+        if (count($qname) === 2) {
+            list($prefix, $element) = $qname;
+        } else {
+            $prefix = null;
+            list($element) = $qname;
+        }
+        $ns = $xml->lookupNamespaceUri($prefix);
+        $handler = Utils::getContainer()->getElementHandler($ns, $element);
```

Comment:
This one is wrong, we should use `getIdentifierHandler()` instead (we have that method precisely for this use case in particular).

Reply:
The `getIdentifierHandler()` only makes sense on BaseID, not on Condition?

Reply:
You're absolutely right!